### ISSUE TITLE
[Question Review] #14 Batch 2A — cue-bias remediation (21 items)

### ISSUE THREAD

**Issue by Naiyi-Chia** [Naiyi-Chia/IPAS-AI-trainer#46]
Parent: #14

## Goal
Human-review and remediate the next scoped set of self-authored questions identified by Issue #14 cue-bias audit.

## Priority
Severe length cue (>=3x)

## Scope
Exactly 21 questions:
- Q085–Q087 — 缺失值與異常值 (L22/L22201; max ratio 3.11x)
- Q112–Q114 — 匿名化與再識別 (L22/L22404; max ratio 3.71x)
- Q127–Q129 — Random Forest (L23/L23202; max ratio 3.78x)
- Q133–Q135 — 特徵工程 (L23/L23301; max ratio 3.13x)
- Q139–Q141 — 交叉驗證 (L23/L23303; max ratio 3.00x)
- Q154–Q156 — RAG 與向量檢索 (L12/L12202; max ratio 3.83x)
- Q169–Q171 — 分類評估 (L23/L23303; max ratio 3.00x)

## Requirements
- Start from the latest `dev` at implementation time; do not use earlier prototype branches as baseline.
- Preserve each question's level / subject / topic / concept / difficulty / answer index / sourceType unless a verified content error is first reflected in this Issue.
- Rewrite distractors to be plausible alternatives at the same semantic level.
- Remove exploitable length / completeness / absurd-distractor / grammar cues.
- Reduce repeated option-pattern memorization within every reviewed cluster.
- Keep one unambiguous correct answer and explanation consistency.
- Do not modify official past-paper content, localStorage behavior, or unrelated UI/code.

## Acceptance Criteria
- [ ] Exactly the scoped IDs change; no other question content changes.
- [ ] Total self-authored question count remains 483 and IDs remain unique.
- [ ] Protected metadata and answer indices remain unchanged unless this Issue is updated first for a verified content error.
- [ ] Every modified item is manually reviewed for one unambiguous correct answer and plausible distractors.
- [ ] No modified item has correct-option / average-distractor length ratio >= 2.0x.
- [ ] Each reviewed option-set cluster is no longer duplicated verbatim across the bank.
- [ ] Audit metrics are re-run and before/after evidence is recorded.
- [ ] JS syntax plus representative Practice / Mock rendering and scoring smoke checks pass.
- [ ] Engineering Ready for Review report is posted before any PR is opened.

## Human Gates
Technical Review → Feature PR to `dev` → Integration Approval → Dev Preview Product Verify. No merge/release without explicit human approval.

## Sequencing
Implement after earlier #14 batches that touch the one-line question DB have been integrated into `dev`, to avoid branch/diff conflicts.

**Comment by Naiyi-Chia** [comment]
## Engineering handoff — ready to implement

Scoped implementation branch created:

- Branch: `question/issue-46-batch2a-remediation`
- Baseline: latest `question/issue-14-remediation`
- Baseline SHA at branch creation: `952c878afb2ffe32e16e3d58f16f33a671ae7e2d`
- PR target after Technical Review: `question/issue-14-remediation`

Engineering Agent / Codex instructions:
1. Fetch origin and check out `question/issue-46-batch2a-remediation`.
2. Read current `AGENTS.md`, `PROJECT_CONTEXT.md`, Parent #14, and this Issue #46.
3. Implement exactly Q085–Q087, Q112–Q114, Q127–Q129, Q133–Q135, Q139–Q141, Q154–Q156, Q169–Q171.
4. Preserve protected metadata and answer indices unless a verified content error is first reflected in Issue #46.
5. Keep distractors plausible and same-level; remove exploitable length/completeness/absurd/grammar cues.
6. Ensure all modified items have correct-option / average-distractor ratio <2.0x and reviewed option-set clusters are no longer verbatim duplicates.
7. Re-run whole-bank cue-bias metrics and update reproducible evidence.
8. Run JS syntax plus representative Practice / Mock render/scoring smoke checks and `git diff --check`.
9. Commit + push only. Do **not** open a PR, merge, or close the Issue.
10. Post `<!-- engineering-ready-for-review -->` completion report to #46 with branch, commit SHA, files changed, QA results, working-tree state, and remote-sync state.

Human Gates remain in force.

**Comment by Naiyi-Chia** [comment]
## Engineering Ready for Review — Issue #46

- Branch: `question/issue-46-batch2a-remediation`
- Commit: `dd77f2c441a21cd7b9de04067e2053227d4181a9`
- Baseline: latest `question/issue-14-remediation` at implementation start, `952c878afb2ffe32e16e3d58f16f33a671ae7e2d`.
- Files: `index.html`, `scripts/audit_question_cues.py`, `scripts/serve_question_batch1a.py`, `docs/QUESTION_CUE_BIAS_BATCH2A.csv`, `docs/QUESTION_CUE_BIAS_BATCH2A.md`.
- Diff: +1548/-14, primarily the before/after CSV and editorial report; the one-line DB is +1/-1.

### Changes

Remediated exactly Q085–Q087, Q112–Q114, Q127–Q129, Q133–Q135, Q139–Q141, Q154–Q156, Q169–Q171. Each was reviewed individually for one correct answer and plausible distractors, with revised stems/options/explanations and differentiated option sets. Protected metadata, answer indices, sourceType, all other questions, non-question DB metadata and application code outside the DB are unchanged. Official content and localStorage behavior are unchanged. Audit/fixture now support `--batch batch2a`; Batch 1A remains default. No new dependency.

### QA

- Exact-scope, metadata, mapping, app-shell and count checks: PASS; 483 questions/483 unique IDs.
- All 21 ratios <2.0 (maximum 1.2245); every modified option set unique across the bank.
- Whole-bank uniquely longest correct: 437→419; ratio >=2: 281→260; ratio >=3: 57→36; repeated groups: 106→99; repeated questions: 473→452. Answer distribution unchanged: A131/B109/C122/D121.
- Node v24.19.0 inline JS syntax parse: PASS. Python syntax and historical Batch 1A default validation: PASS.
- 375×812 Practice fixture: all 21 correct → 21/21, 100%; one wrong choice per item → 0/21, 0%. Feedback/explanations/navigation passed; no overflow across all 42 answered states.
- Wrong-question view listed 21 items; weak-area counts matched.
- 1440×900 Mock fixture: L12 3/3, L22 6/6, L23 12/12, each scored 100 and opened review; no question overflow; console error log empty.
- Mobile L22 Mock screenshot checked for wrapping and usable controls.
- Unfiltered app: 483 questions; all seven panels opened; Practice start/answer/explanation/next passed (sample included Q154). 50-question Mock submitted with 1 correct/49 blank, scored 2, and entered review. Mobile panels/review had no overflow; console error log empty.
- Cached official 115 second-session L11 opened with 50 questions and official-source labeling.
- `git diff --check`: PASS.

[Full item-by-item editorial review and QA report](https://github.com/Naiyi-Chia/IPAS-AI-trainer/blob/dd77f2c441a21cd7b9de04067e2053227d4181a9/docs/QUESTION_CUE_BIAS_BATCH2A.md) · [Before/after CSV](https://github.com/Naiyi-Chia/IPAS-AI-trainer/blob/dd77f2c441a21cd7b9de04067e2053227d4181a9/docs/QUESTION_CUE_BIAS_BATCH2A.csv)

```powershell
python scripts/audit_question_cues.py --batch batch2a --baseline 952c878afb2ffe32e16e3d58f16f33a671ae7e2d --csv docs/QUESTION_CUE_BIAS_BATCH2A.csv
python scripts/serve_question_batch1a.py --batch batch2a
```
Fixture: `http://127.0.0.1:8766/batch2a`.

### Limitations / handoff

Engineering Agent editorial review does not replace independent Technical Review or Human Product Verify. Heuristics do not establish learner difficulty/discrimination; 260 length candidates and 99 repeated groups remain for later batches. Fixture narrows the pool, uses memory storage and auto-confirms submission. Physical mobile Safari/accessibility, persistence and external PDF ingestion were not retested. Unfiltered submit reported a browser transport timeout, but completed result/review were verified; native confirmation interaction was not independently observed. Newer dev changes remain outside this epic-based scope.

Final working tree clean. Push succeeded; local HEAD matches `origin/question/issue-46-batch2a-remediation`, 0 ahead/0 behind. Future PR target is `question/issue-14-remediation`, subject to review and human authorization. No PR opened, merge performed, Issue closed, Product Verify declared, or release performed.

**Comment by Naiyi-Chia** [comment]
## ChatGPT Technical Review — Issue #46 Batch 2A: PASS

Reviewed branch `question/issue-46-batch2a-remediation` at `dd77f2c441a21cd7b9de04067e2053227d4181a9` against target `question/issue-14-remediation@952c878afb2ffe32e16e3d58f16f33a671ae7e2d`.

### Independent scope / invariant checks
- Branch is ahead 1 / behind 0 vs the epic target.
- Changed question IDs are exactly:
  `Q085–Q087, Q112–Q114, Q127–Q129, Q133–Q135, Q139–Q141, Q154–Q156, Q169–Q171`.
- 483 questions remain; 483 IDs remain unique.
- No protected metadata or answer-index differences found for the 21 reviewed items.
- Non-question DB metadata is unchanged.
- HTML/CSS/JS shell outside the parsed DB is unchanged.
- Answer-position distribution remains A131 / B109 / C122 / D121.

### Independent cue metrics
Recomputed from the baseline and reviewed branch:
- uniquely-longest correct: 437 → 419
- correct/avg-distractor ratio >=2: 281 → 260
- ratio >=3: 57 → 36
- repeated option-set groups: 106 → 99
- questions in repeated sets: 473 → 452
- all 21 reviewed items are <2.0x; max = Q171 at 1.2245x
- all 21 reviewed option sets are unique across the bank

### Content review
Manual review of all 21 rewritten items found no blocking ambiguity or answer/explanation mismatch.

Authoritative spot-checks support the reviewed concepts:
- Random Forest bootstrapping / random feature subsets / variance reduction: https://scikit-learn.org/1.8/modules/ensemble.html
- One-hot encoding for categorical features: https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder
- K-fold semantics: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold
- De-identification / re-identification risk: https://csrc.nist.gov/pubs/sp/800/188/final
- Vector semantic retrieval: https://learn.microsoft.com/en-us/azure/search/vector-search-overview
- F1 definition: https://scikit-learn.org/1.7/modules/generated/sklearn.metrics.f1_score.html

Specific review notes:
- Q085–Q087 correctly distinguish diagnosis/source verification from prematurely deleting/imputing/transformation.
- Q112–Q114 correctly distinguish re-identification/linkage risk from encryption, integrity, availability, or superficial release controls.
- Q127–Q129 correctly distinguish Random Forest from boosting and explain decorrelation/averaging.
- Q133–Q135 correctly constrain one-hot encoding scenarios and preserve training/inference column mapping.
- Q139–Q141 correctly describe K-fold use, fold membership, and interpretation of cross-fold variability without using the final test set for selection.
- Q154–Q156 correctly separate embedding, vector-index retrieval, and generation roles.
- Q169–Q171 correctly define and calculate F1; Q169 gives 0.60 for P=.75/R=.50, and Q170's all-negative case yields F1=0 via the TP/FP/FN form.

### Engineering / QA evidence review
Engineering evidence covers:
- exact-scope / mapping / count assertions
- JS and Python syntax checks
- Batch 1A regression/default behavior
- 375×812 Practice scoring/rendering for all 21 items
- wrong-question / weak-area behavior
- desktop and mobile Mock fixture checks
- unfiltered 483-question Practice / Mock smoke
- cached official-paper smoke
- `git diff --check`

The reported browser transport timeout during unfiltered Mock submit is not a blocker because the completed result/review state was subsequently verified and the fixture submit/scoring path also passed. Physical Safari/accessibility and external PDF ingestion remain outside this Technical Review and are not required for epic-branch accumulation.

### Result
**PASS — no blocking findings.**

This authorizes moving to a Sub-issue PR targeting `question/issue-14-remediation`. Merge still requires explicit **Human Epic Integration Approval**. This is not canonical Integration Approval, Product Verify, or release approval.

**Comment by Naiyi-Chia** [comment]
Feature PR: #75
Head: `question/issue-46-batch2a-remediation`
Base: `question/issue-14-remediation`
Status: Open — waiting for Human Epic Integration Approval

**Comment by Naiyi-Chia** [comment]
## Epic integration update

PR #75 has been squash-merged into `question/issue-14-remediation` after explicit Human Epic Integration Approval.

- Merge commit: `0cebcdfc7e93f6b9ca581f87b3345f8c23b09f79`
- Technical Review: PASS
- Human Epic Integration Approval: PASS
- Canonical Integration Approval: not yet applicable
- Product Verify: not yet applicable
- Release Approval: not applicable

Issue #46 remains open as part of Parent #14 until the epic completes final integration through `dev`, Product Verify, release, and Production Smoke.
